x-pack/plugins/stack_connectors/server/connector_types/microsoft_defender_endpoint/microsoft_defender_endpoint.ts

```diff
@@ -0,0 +1,219 @@
+    });
+
+    this.registerSubAction({
+      name: MICROSOFT_DEFENDER_ENDPOINT_SUB_ACTION.RELEASE_HOST,
+      method: 'releaseHost',
+      schema: ReleaseHostParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: MICROSOFT_DEFENDER_ENDPOINT_SUB_ACTION.TEST_CONNECTOR,
+      method: 'testConnector',
+      schema: TestConnectorParamsSchema,
+    });
+  }
+
+  private async fetchFromMicrosoft<R extends MicrosoftDefenderEndpointBaseApiResponse>(
+    req: Omit<SubActionRequestParams<R>, 'responseSchema'>,
+    connectorUsageCollector: ConnectorUsageCollector
+  ): Promise<R> {
+    const bearerAccessToken = await this.oAuthToken.get(connectorUsageCollector);
+    const response = await this.request<R>(
+      {
+        ...req,
+        // We don't validate responses from Microsoft API's because we do not want failures for cases
+        // where the external system might add/remove/change values in the response that we have no
+        // control over.
+        responseSchema:
+          MicrosoftDefenderEndpointDoNotValidateResponseSchema as unknown as SubActionRequestParams<R>['responseSchema'],
+        headers: { Authorization: `Bearer ${bearerAccessToken}` },
+      },
+      connectorUsageCollector
+    );
+
+    return response.data;
+  }
+
+  protected getResponseErrorMessage(error: AxiosError): string {
+    const appendResponseBody = (message: string): string => {
+      const responseBody = JSON.stringify(error.response?.data ?? {});
+
+      if (responseBody) {
+        return `${message}\nURL called: ${error.response?.config?.url}\nResponse body: ${responseBody}`;
+      }
+
+      return message;
+    };
+
+    if (!error.response?.status) {
+      return appendResponseBody(error.message ?? 'Unknown API Error');
+    }
+
+    if (error.response.status === 401) {
+      return appendResponseBody('Unauthorized API Error (401)');
+    }
+
+    return appendResponseBody(`API Error: [${error.response?.statusText}] ${error.message}`);
+  }
+
+  public async testConnector(
+    _: MicrosoftDefenderEndpointTestConnectorParams,
+    connectorUsageCollector: ConnectorUsageCollector
+  ): Promise<MicrosoftDefenderEndpointTestConnector> {
+    const results: string[] = [];
+
+    const catchErrorAndIgnoreExpectedErrors = (err: Error) => {
+      if (err.message.includes('ResourceNotFound')) {
+        return '';
+      }
+      throw err;
+    };
+
+    await this.getAgentDetails({ id: 'elastic-connector-test' }, connectorUsageCollector)
+      .catch(catchErrorAndIgnoreExpectedErrors)
+      .then(() => {
+        results.push('API call to Machines API was successful');
+      });
+
+    await this.isolateHost(
+      { id: 'elastic-connector-test', comment: 'connector test' },
+      connectorUsageCollector
+    )
+      .catch(catchErrorAndIgnoreExpectedErrors)
+      .then(() => {
+        results.push('API call to Machine Isolate was successful');
+      });
+
+    await this.isolateHost(
```

Comment:
not releaseHost ?

Reply:
Good catch. Thank you. I will adjust it.